### PR TITLE
Duplo 13405 Terraform provider - duplocloud batch job definitions with revisions do not delete correctly

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Bug fix for Terraform gets timed out while destroying batch job resources

## Summary of changes
Terraform Job Deletion time out issue fixed
This PR does the following:

- When a change is made to the batch job that requires recreation, terraform  deletes all the revisions of the job and then recreate it.
 

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
bug_fix, enhancement

___

## **Description**
- Implemented `duplocloud_gcp_node_pools` resource in the provider, allowing management of GCP Kubernetes node pools.
- Fixed deletion of AWS Batch Job Definitions to correctly handle all revisions.
- Added SDK functions for creating, reading, updating, and deleting GCP K8 Node Pools.
- Updated CHANGELOG with recent fixes and enhancements.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>provider.go</strong><dd><code>Add GCP Node Pools Resource to Provider</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/provider.go

- Added `duplocloud_gcp_node_pools` to the provider's resources.

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/425/files#diff-0959744faba70cfc67abd70733e39a6f668d41c9fae69a27968841ef8ed56eeb">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_gcp_k8_node_pool.go</strong><dd><code>Implement GCP K8 Node Pool Resource</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_gcp_k8_node_pool.go

<li>Implemented a new resource <code>duplocloud_gcp_k8_node_pool</code> with full CRUD <br>operations.<br> <li> Added various configurations for GCP K8s node pool including <br>auto-scaling, upgrade settings, and node taints.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/425/files#diff-bc0cc4a27038eb399598c526fd7148bd7cd9349310b06235a990a5f2cac1eaf9">+735/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>aws_batch.go</strong><dd><code>Add Function to Get All Revisions of AWS Batch Job Definition</code></dd></summary>
<hr>
      
duplosdk/aws_batch.go

<li>Added <code>AwsBatchJobDefinitionGetAllRevisions</code> to fetch all revisions of a <br>batch job.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/425/files#diff-7ed1bf367906c422ef26cf3db813258caa061b4df68a1c9a6d7c2a65d7dd65c1">+16/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_node_pools.go</strong><dd><code>Add SDK Support for GCP K8 Node Pools</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplosdk/gcp_node_pools.go

<li>Added SDK support for GCP K8 Node Pools including create, read, <br>update, and delete operations.<br> <li> Included structures for node pool configurations like auto-scaling, <br>logging, and accelerators.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/425/files#diff-d7d6988aa1383fd5e2f3442050fe77b783a7d1a7094e70e20982f64d9f1c5b03">+129/-0</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Bug_fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_aws_batch_job_definition.go</strong><dd><code>Fix AWS Batch Job Definition Deletion to Handle All Revisions</code></dd></summary>
<hr>
      
duplocloud/resource_duplo_aws_batch_job_definition.go

<li>Modified <code>resourceAwsBatchJobDefinitionDelete</code> to delete all revisions <br>of a batch job.<br> <li> Removed revision specific deletion in favor of deleting all revisions.<br> <br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/425/files#diff-840f1caa3a415f3c37a1919211ba5fe6ffc313f694e092f17045e8374627eec9">+9/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update CHANGELOG for AWS Batch Job and GCP Node Pools Changes</code></dd></summary>
<hr>
      
CHANGELOG.md

<li>Updated CHANGELOG to include fixes and additions related to AWS Batch <br>Job Definitions and GCP Node Pools.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/425/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+11/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-13405 Terraform provider - duplocloud batch job definitions with revisions do not delete correctly](https://app.clickup.com/t/8655600/DUPLO-13405)

**Review by joek-duplo — CHANGES_REQUESTED** [review]
This PR should not contain any changes for nodepools at all.   Please remove that and fix the changelog.

File: duplocloud/resource_duplo_aws_batch_job_definition.go [hunk 1]
```diff
@@ -289,20 +289,21 @@ func resourceAwsBatchJobDefinitionDelete(ctx context.Context, d *schema.Resource
...
 	}
-
-	clientErr := c.AwsBatchJobDefinitionDelete(tenantID, fullName+":"+strconv.Itoa(revision))
-	if clientErr != nil {
-		if clientErr.Status() == 404 {
-			return nil
+	for _, data := range *jq {
+		clientErr := c.AwsBatchJobDefinitionDelete(tenantID, fullName+":"+strconv.Itoa(data.Revision))
+		if clientErr != nil {
+			if clientErr.Status() == 404 {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_aws_batch_job_definition.go:302]: We should just continue the loop in case of 404, for deletion of another revision here.

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_aws_batch_job_definition.go:302]: made changes
resolved conflicts

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**